compound.hh

```diff
@@ -117,15 +118,15 @@ public:
     managed_bytes serialize_optionals(std::span<const bytes_opt> values) const {
         return serialize_value(boost::make_iterator_range(values.begin(), values.end()) | boost::adaptors::transformed([] (const bytes_opt& bo) -> bytes_view {
             if (!bo) {
-                throw std::logic_error("attempted to create key component from empty optional");
+                throw exceptions::invalid_request_exception("Invalid null value in condition for column {}");
             }
             return *bo;
         }));
     }
     managed_bytes serialize_optionals(std::span<const managed_bytes_opt> values) const {
         return serialize_value(boost::make_iterator_range(values.begin(), values.end()) | boost::adaptors::transformed([] (const managed_bytes_opt& bo) -> managed_bytes_view {
             if (!bo) {
-                throw std::logic_error("attempted to create key component from empty optional");
+                throw exceptions::invalid_request_exception("Invalid null value in condition for column {}");
```

Comment:
Same here. This should not be changed. Not only is the exception type not necessarily the right one, the text is quite likely to not be right right one (why "condition"). The optional can and should be checked in the caller - see my comment above. The "logic_error" is correct, the from_optional_exploded() function even says in a comment:
```c++
    // We don't allow optional values, but provide this method as an efficient adaptor
```
So calling it for an unset optional is a bug.